### PR TITLE
Validate max amount. Add error logs.

### PR DESCRIPTION
**Pull request by jbguardian**
The scope of this PR must include:
- validate max amount for endpoints - /contribute/one-off/paypal/create-payment and /contribute/one-off/stripe/execute-payment.
- Adding error logs and payment was not included in the DB and no error were displayed in the file.

### CODE REVIEW

File: src/main/scala/model/Currency.scala [hunk 1]
```diff
@@ -22,4 +22,11 @@ object Currency extends Enum[Currency] with CirceEnum[Currency] {
...
 
   case object NZD extends Currency
+
+  def isMaxAmount(amount: Int, currency: Currency): Boolean = {
+    val maxAmount = if (currency.equals(AUD)) 16000 else 2000
+    if (amount < maxAmount) false else true
+  }
 }
+
+
```
Comment (jranks123) [inline src/main/scala/model/Currency.scala]: No need for these extra lines

File: src/main/scala/model/Currency.scala [hunk 2]
```diff
@@ -22,4 +22,11 @@ object Currency extends Enum[Currency] with CirceEnum[Currency] {
   case object EUR extends Currency
 
   case object NZD extends Currency
+
+  def isMaxAmount(amount: Int, currency: Currency): Boolean = {
+    val maxAmount = if (currency.equals(AUD)) 16000 else 2000
+    if (amount < maxAmount) false else true
```
Comment (jranks123) [inline src/main/scala/model/Currency.scala]: I think this should be <=

File: src/main/scala/model/Currency.scala (hunk 1 above)
Comment (jbguardian) [reply src/main/scala/model/Currency.scala]: 👍

File: src/main/scala/model/Currency.scala (hunk 2 above)
Comment (jbguardian) [reply src/main/scala/model/Currency.scala]: true, max amount included. 👍

File: src/main/scala/services/PaypalService.scala [hunk 3]
```diff
@@ -123,14 +133,20 @@ class PaypalService(config: PaypalConfig)(implicit pool: PaypalThreadPool) exten
...
+      logger.error(s"Error getting capture from related resources. Error: $error")
+      PaypalApiError.fromThrowable(error)
+    }
 
 
   private def validateCapture(capture: Capture): Either[PaypalApiError, Capture] = {
     if (capture.getState.toUpperCase.equalsIgnoreCase("COMPLETED"))
       Right(capture)
-    else
+    else{
```
Comment (joelochlann) [inline src/main/scala/services/PaypalService.scala]: `else {`

File: src/main/scala/services/PaypalService.scala [hunk 4]
```diff
@@ -123,14 +133,20 @@ class PaypalService(config: PaypalConfig)(implicit pool: PaypalThreadPool) exten
   private def getCapture(relatedResources: RelatedResources, transaction: Transaction): Either[PaypalApiError, Capture] =
     Either.catchNonFatal(relatedResources
       .getAuthorization
-      .capture(apiContext, buildCaptureByTransaction(transaction))).leftMap(PaypalApiError.fromThrowable)
+      .capture(apiContext, buildCaptureByTransaction(transaction))).leftMap{ error =>
+      logger.error(s"Error getting capture from related resources. Error: $error")
+      PaypalApiError.fromThrowable(error)
+    }
```
Comment (joelochlann) [inline src/main/scala/services/PaypalService.scala]: Indentation here is confusing. What about:

```scala
Either.catchNonFatal(
  relatedResources
    .getAuthorization
    .capture(apiContext, buildCaptureByTransaction(transaction))
).leftMap { error =>
  logger.error(s"Error getting capture from related resources. Error: $error")
  PaypalApiError.fromThrowable(error)
}
    
```

File: src/main/scala/model/Currency.scala [hunk 5]
```diff
@@ -22,4 +22,9 @@ object Currency extends Enum[Currency] with CirceEnum[Currency] {
   case object EUR extends Currency
 
   case object NZD extends Currency
-}
+
+  def isMaxAmount(amount: Int, currency: Currency): Boolean = {
```
Comment (joelochlann) [inline src/main/scala/model/Currency.scala]: I think `exceedsMaxAmount` would be a better name

File: src/main/scala/services/PaypalService.scala (hunk 3 above)
Comment (jbguardian) [reply src/main/scala/services/PaypalService.scala]: 👍

File: src/main/scala/services/PaypalService.scala (hunk 4 above)
Comment (jbguardian) [reply src/main/scala/services/PaypalService.scala]: 👍

File: src/main/scala/model/Currency.scala (hunk 5 above)
Comment (jbguardian) [reply src/main/scala/model/Currency.scala]: 👍

**Pull request merged**